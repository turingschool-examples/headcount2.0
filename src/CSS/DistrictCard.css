.year-red {
  color: red;
}

.year-black {
  color: green;
}

.district-card {
  text-align: center;
  box-shadow: 2px 2px 2px 2px #ccc;
}

.district-location {
  font-size: 1.5rem;
}
