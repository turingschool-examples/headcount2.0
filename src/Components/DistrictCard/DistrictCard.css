.highStats {
  color: #7097B2;
}

.lowStats {
  color: #B81F38;
}

.card {
  border-radius: 18px;
  border: solid #4F4C47 2px;
  color: #4F4C47;
  display: inline-block;
  margin: 1rem;
  padding: .5rem;
  width: 300px;
}

.selected {
  background-color: #F7F5F1;
  border: solid #21201E 4px;
}

.card:hover {
  border: solid #4F4C47 4px;
}

.district {
  font-family: avenir, sans-serif;
  font-size: 1.5em;
  text-align: center;
}

.stats {
  font-family: avenir, sans-serif;
  font-size: 1.25em;
  padding-left: 5rem;
}