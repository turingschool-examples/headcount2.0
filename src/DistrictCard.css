.districtCard {
  box-shadow: 1em;
  border-radius: 15px;
  background-color: white;
  display: flex;
  flex-direction: column;
  width: 15rem;
  height: auto;
  margin: 0 0 5rem 0;
  padding-bottom: 1rem;
  text-align: center
}

.districtLocation {
  font-size: 1rem
}

.districtStats {
  list-style-type: none;
  text-align: center;
  margin: 0;
  padding: 0
}

.statGreater {
  color: blue
}

.statLesser {
  color: red
}