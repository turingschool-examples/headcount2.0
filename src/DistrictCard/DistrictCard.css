.DistrictCard {
	display: flex;
	align-items: center;
	flex-direction: column;
	justify-content: center;
	background-color: #FFF;
	text-align: center;
	width: 15rem;
	height: 26rem;
	margin: 0 auto 0.5rem;
	padding: 0 0 1rem;
	box-shadow: 0px 1px 1px 2px #F5F6F7;
	border-radius: 10px;
	font-family: 'Open Sans', sans-serif;
}

.card-location {
	text-align: center;
	overflow-wrap: break-word;
	margin: 1.5rem 1.5rem 0.75rem;
	font-family: 'Arvo', serif;
	font-size: 1.25rem;
}

/*.card-stats {
	margin: 0 auto 1.5rem;
}
*/
.greater-than-point-5 {
	color: #01411b;
	background-color: #F5F6F7;
	margin: 0.2rem auto;
	font-weight: 300;
	padding: 0 0.75rem;
	border-radius: 10px;
}

.less-than-point-5 {
	color: #FFF;
	background-color: #008b38;
	margin: 0.2rem auto;
	font-weight: 300;
	padding: 0 0.75rem;
	border-radius: 10px;
}


