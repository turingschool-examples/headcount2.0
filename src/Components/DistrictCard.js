import React from "react";
import "../CSS/DistrictCard.css";
import PropTypes from "prop-types";

export const DistrictCard = ({
  location,
  stats,
  clicked,
  selectCard,
  counter
}) => {
  return (
    <div
      onClick={() => selectCard(location)}
      className={clicked ? "district-card-clicked" + counter : "district-card"}
    >
      <h1 className="district-location">{location}</h1>
      <header className={clicked ? "year-header-show" : "year-header-hide"}>
        Year
      </header>
      <header className={clicked ? "avg-header-show" : "avg-header-hide"}>
        Avg
      </header>
      {Object.keys(stats).map((year, index) => {
        const avg = stats[year];
        return (
          <section
            className={clicked ? "stats-show" : "stats-hide"}
            key={index}
          >
            <p className="year">
              {year}
              ......
              <span className={avg < 0.5 ? "avg-low" : "avg-high"}>{avg}</span>
            </p>
          </section>
        );
      })}
      <p className={clicked ? "show-msg" : "hide-msg"}>Click to Remove</p>
    </div>
  );
};

const { string, shape, number, bool, func } = PropTypes;

DistrictCard.propTypes = {
  location: string,
  stats: shape({ year: number }),
  clicked: bool,
  selectCard: func,
  counter: number
};
