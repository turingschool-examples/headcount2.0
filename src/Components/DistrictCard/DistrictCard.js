import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';


const DistrictCard = ({district , selectDistrict}) => {
  const years = Object.keys(district.stats)

  const determineColor = (stats) => {
    return stats > 0.5 ? "highStats" : "lowStats"
  }

  const createStats = years.map((year, index) => {
    const stats = district.stats[year]
    return (
      <li className={determineColor(stats)} key={index}>
        {[year]}: {stats}
      </li>
      )
  })

  const handleClick = (location) => {
    selectDistrict(location)
  }

  return (
    <section 
      className={`card ${district.selected ? 'selected' : ''}`} 
      id={district.key}
      key={district.key}
      onClick={() => {handleClick(district.location)}}
    > 
      <h2 className="district">District Name: {district.location}</h2>
      <ul className="stats">
        {createStats}
      </ul>
    </section>
  )
}

DistrictCard.propTypes = { 
  district: PropTypes.object
}



export default DistrictCard;