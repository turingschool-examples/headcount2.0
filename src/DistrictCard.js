import React from 'react'
import './DistrictCard.css'

export const DistrictCard = ({location, stats, id, selectCard, deleteCard}) => {
  // console.log(districts)
  const displayStats = Object.entries(stats).map((stat) => {
    return  <li className={stat[1] > .5 ? "statGreater" : "statLesser"}>{stat[0]}: {stat[1]}</li>})

  return (
    <div className="districtCard" key={id} onClick={(e)=>{selectCard(location)}}>
      <h1 className="districtLocation">{location}</h1>
      <ul className="districtStats">{displayStats}</ul>
    </div>
  )
}