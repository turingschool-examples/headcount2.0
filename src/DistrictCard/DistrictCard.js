import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';

const DistrictCard = ({ location, stats, district, compareDistrict }) => {

	const statsKeys = Object.keys(stats)
	const schoolData = statsKeys.map((stat) => {
		// console.log(stat)
		return <p 
			key={Math.random()}
			className={(stats[stat] > 0.5) ? 'greater-than-point-5' : 'less-than-point-5'}>
			{stat}: {stats[stat]} 
		</p>
	})
	
if (compareDistrict) {
		return (
		<div className='DistrictCard' onClick={ () => compareDistrict(district) }>
			<h3 className='card-location'>{location}</h3>
			{schoolData}
		</div>
	)
	} else {
		return (
			<div className='DistrictCard'>
				<h3 className='card-location'>{location}</h3>
				{schoolData}
			</div>
		)
	}

}

DistrictCard.proptypes = {
	location: PropTypes.string.isRequired,
	stats: PropTypes.object.isRequired,
	district: PropTypes.object.isRequired,
	compareDistrict: PropTypes.func
}

export default DistrictCard;
