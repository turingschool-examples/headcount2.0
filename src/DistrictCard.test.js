import React from 'react';
import { shallow } from 'enzyme';
import { DistrictCard } from './DistrictCard.js'

describe('DistrictCard', () => {

  it('should match the snapshot', () => {
    const mockProp = {
                      location: 'alabama',
                      stats: {2014: 0}
    }
    const wrapper = shallow(<DistrictCard location={mockProp.location} stats={mockProp.stats}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a different class name for stats > .5', () => {
    const mockProp = {
                      location: 'alabama',
                      stats: {2014: .6}
    }   
    const wrapper = shallow(<DistrictCard location={mockProp.location} stats={mockProp.stats}/>)
    expect(wrapper.find('li').hasClass('statGreater'))
  })

  it('should have a different class name for stats < .5', () => {
    const mockProp = {
                      location: 'alabama',
                      stats: {2014: .3}
    }   
    const wrapper = shallow(<DistrictCard location={mockProp.location} stats={mockProp.stats}/>)
    expect(wrapper.find('li').hasClass('statLesser'))
  })

})