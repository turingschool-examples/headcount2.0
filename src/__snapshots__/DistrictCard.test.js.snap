// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    location="alabama"
    stats={
        Object {
            "2014": 0,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="districtLocation"
>
          alabama
</h1>,
        <ul
          className="districtStats"
>
          <li
                    className="statLesser"
          >
                    2014
                    : 
                    0
          </li>
</ul>,
      ],
      "className": "districtCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "alabama",
          "className": "districtLocation",
        },
        "ref": null,
        "rendered": "alabama",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="statLesser"
>
              2014
              : 
              0
</li>,
          ],
          "className": "districtStats",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2014",
                ": ",
                0,
              ],
              "className": "statLesser",
            },
            "ref": null,
            "rendered": Array [
              "2014",
              ": ",
              0,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="districtLocation"
>
            alabama
</h1>,
          <ul
            className="districtStats"
>
            <li
                        className="statLesser"
            >
                        2014
                        : 
                        0
            </li>
</ul>,
        ],
        "className": "districtCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "alabama",
            "className": "districtLocation",
          },
          "ref": null,
          "rendered": "alabama",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="statLesser"
>
                2014
                : 
                0
</li>,
            ],
            "className": "districtStats",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2014",
                  ": ",
                  0,
                ],
                "className": "statLesser",
              },
              "ref": null,
              "rendered": Array [
                "2014",
                ": ",
                0,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
